nilearn/interfaces/fmriprep/load_confounds_utils.py

```diff
@@ -230,9 +230,9 @@ def _get_file_name(nii_file):
     # https://www.geeksforgeeks.org/python-sort-list-of-lists-by-the-size-of-sublists/
     confound_file_candidates = sorted(confound_file_candidates, key=len)[::-1]
     confound_file_candidates = [
-        os.path.join(base_dir, crc) for crc in confound_file_candidates
+        base_dir / crc for crc in confound_file_candidates
     ]
-    found_files = [cr for cr in confound_file_candidates if os.path.isfile(cr)]
+    found_files = [str(cr) for cr in confound_file_candidates if cr.is_file()]
```

Comment:
Do we want to return a `Path` here since this function is private?

Reply:
I would say so: though once again, depends how many of our things break.